c7n/resources/s3.py

```diff
@@ -2203,3 +2203,87 @@ def process_chunk(self, batch, bucket):
             Bucket=bucket['Name'], Delete={'Objects': objects}).get('Deleted', ())
         if self.get_bucket_style(bucket) != 'versioned':
             return results
+
+
+@actions.register('configure-lifecycle')
+class Lifecycle(BucketActionBase):
+    """Action applies a lifecycle policy to versioned S3 buckets
+
+    The schema to supply to the rule follows the schema here: goo.gl/yULzNc
+
+    :example:
+
+        .. code-block: yaml
+
+            policies:
+              - name: s3-apply-lifecycle
+                resource: s3
+                actions:
+                  - type: configure-lifecycle
+                    rules:
+                      - ID: my-lifecycle-id
+                        Status: Enabled
+                        Prefix: foo/
+                        Transitions:
+                          - Days: 60
+                            StorageClass: GLACIER
+
+    """
+
+    schema = type_schema(
+        'configure-lifecycle',
+        **{
+            'rules': {
+                'type': 'array',
+                'required': True,
+                'items': {
+                    'type': 'object',
+                    'required': ['ID', 'Status'],
+                    'properties': {
```

Comment:
open question? thoughts on having the actual lifecycle rule schema inline here, instead of partial?

Reply:
@kapilt - My hesitation on that was just that when it meant we have to track that schema and be in sync with it.  I do think it is more user friendly to have it, as long as you are ok with supporting it in the future.  Let me know if you'd like me to add it.

Reply:
i'm okay with it its relatively static, basically add only due to compatibility concerns.